### PR TITLE
Update release notes of js provider v1

### PR DESCRIPTION
**Pull request by Eskibear**

### CODE REVIEW

File: releaseNotes/JavaScriptProvider.md [hunk 1]
```diff
@@ -2,6 +2,15 @@
 
 [Source code][source_code] | [Package (npm)][package] | [Samples][samples]
 
+## 1.0.0 - June 6, 2024
+
+**Release of the stable API version `v1`**.
```
Comment (avanigupta) [inline releaseNotes/JavaScriptProvider.md:7]: Usually we use the following format for stable version update:
`This is the first stable release of XYZ feature introduced in ABC version`

In this case, I think we can say that this is the first stable release of all the features that were released in previous preview versions.

Comment (zhenlan) [reply releaseNotes/JavaScriptProvider.md:7]: I think we should relist all features that made it to the stable release from the previous preview releases.

Comment (Eskibear) [reply releaseNotes/JavaScriptProvider.md]: updated. listed key features, and added a link to quickstart. See if we want to polish the wording or provide more details about the features.

File: releaseNotes/JavaScriptProvider.md [hunk 2]
```diff
@@ -2,6 +2,24 @@
...
 
+## 1.0.0 - June 6, 2024
+
+This is the first stable release of the following features.
+- Loading and composing configuration with key-value selectors.
+- Authenticating with connection string or Microsoft Entra Id.
+- Loading key-values as either a `Map` or an object based on needs.
+- Trimming prefixes from keys.
+- Loading key-values with JSON content type direct as an object.
+- Resolving Key Vault reference.
```
Comment (zhenlan) [inline releaseNotes/JavaScriptProvider.md:13]: "Resolving Key Vault reference**s**" or "Key Vault reference resolution"

**Review by zhenlan — APPROVED** [review]

File: releaseNotes/JavaScriptProvider.md (hunk 2 above)
Comment (Eskibear) [reply releaseNotes/JavaScriptProvider.md]: updated to "Key Vault reference resolution"

File: releaseNotes/JavaScriptProvider.md [hunk 3]
```diff
@@ -2,6 +2,24 @@
 
 [Source code][source_code] | [Package (npm)][package] | [Samples][samples]
 
+## 1.0.0 - June 6, 2024
+
+This is the first stable release of the following features.
+- Loading and composing configuration with key-value selectors.
+- Authenticating with connection string or Microsoft Entra Id.
+- Loading key-values as either a `Map` or an object based on needs.
```
[the remark this answers is not in the document]
Comment (zhenlan) [reply releaseNotes/JavaScriptProvider.md]: I thought we called it "configuration object" in our document.

[the remark this answers is not in the document]
Comment (Eskibear) [reply releaseNotes/JavaScriptProvider.md]: my bad, a miss. Thanks for the double check, fixed in 7d439c1 now.

**Review by zhenlan — APPROVED** [review]

**Pull request merged**